draft-pauly-add-requirements.md

```diff
@@ -186,68 +186,73 @@ Using a content-provider's encrypted resolver can also provide several privacy a
 
-# Distilled discovery mechanisms {#mechanisms}
+# Discovery mechanisms {#mechanisms}
 
-The various use cases described in {{use-cases}} do not all necessarily require separate mechanisms.
+The use cases described in {{use-cases}} do not all necessarily require separate mechanisms.
 
-Generally, the use cases can be summarized in two catgories:
+Generally, the use cases can be summarized in two categories:
 
-1. Discover encrypted resolvers equivalent to (or associated with) unencrypted resolvers. (Network-provisioned,
-client-selected, VPN)
-2. Discover encrypted resolvers applicability to a limited set of domains. (Private names, local content, CDN content)
+1. Encrypted resolver discovery: Discover encrypted resolvers equivalent to (or associated with) unencrypted resolvers.
+Examples include network-provisioned, client-selected, and VPN-configured resolvers.
+2. Domain-specific encrypted resolver discovery: Discover encrypted resolvers applicable to a limited set of domains.
+Examples include resolvers for enterprise or private names, local content, and CDN content.
 
-For the first group, mechanisms to discover equivalent resolvers can either add new parameters to existing provisioning
+Encrypted resolver discovery mechanisms can either add new parameters to existing provisioning
 mechanisms (adding necessary information to use DoT or DoH to options in DHCP, RAs, or IKEv2) or else provide a way
 to communicate with a provisioned unencrypted DNS resolver and discover the equivalent or associated encrypted
 DNS resolver.
 
-For the second group, mechanisms additionally need to provide some information about the applicability and capabilities
-of encrypted resolvers. This information could either be provisioned,
+Domain-specific encrypted resolver discovery mechanisms additionally need to provide some information about the
+applicability and capabilities of encrypted resolvers. This information could either be provisioned,
 or be discovered based on clients actively trying to access content.
 
 # Privacy and security requirements {#priv-sec}
 
-Since one of the main reasons for enabling encrypted DNS on clients is to improve the privacy and security properties
-of DNS queries, discovery mechanisms must be designed carefully to not weaken a client's security or privacy.
+A primary goal of encrypted DNS is improving the privacy and security of DNS queries and answers in the presence
+of malicious attackers. Such attackers are assumed to interfere with or otherwise impede DNS traffic and corresponding
+discovery mechanisms, in any way possible, in an attempt to prevent usage of encrypted DNS. For example, an attacker
+may block connectivity to a resolver chosen by the user. Clients MUST treat the network as such an attacker unless given
+some means of authenticating or otherwise trusting the network.
 
-- Clients are always responsible for selecting which resolver(s) to use. Standards describing resolver discovery
```

Comment:
I don't think we should lose these points. These are important for the conversation and come from MT's email points.

Reply:
I guess that's covered as "Override or adversely influence client resolver selection by users or administrators"? Seems a bit less clear?

Reply:
Yeah, I rephrased it in that bullet. Saying that clients get to pick at the end of the day is table stakes, and it seems the attacker should never be able to change that. (Hence why I wrote it this way.)